app/controllers/authentications/microsoft_graph_oauth_controller.rb

```diff
@@ -0,0 +1,96 @@
+class Authentications::MicrosoftGraphOauthController < ApplicationController
+  allow_unauthenticated_access
+
+  def create
+    if Current.user
+      Current.user.microsoft_graph_credential&.destroy
+      _, cred = add_person_credentials("MicrosoftGraphCredential")
+      cred.save! && redirect_to(edit_settings_person_path, notice: "Saved") && return
+
```

Comment:
I don't think you're ever going to hit this first conditional, are you? The Google Oauth method has this condition because it's used for two things: logging in and for giving the AI access to your google inbox, calendar, tasks. But right now Microsoft is only used for logging in.

If there is a Current.user, you'll never be able to see the login page, so you'll never be able to submit to this `create` endpoint, right?

Reply:
"Never" is strong. Someone might have the Login page open on two tabs (say after being automatically logged out overnight or similar), and come back and click Login on each one.

I'll update this to at least bail out of the `create` if the user is already logged in:

```ruby
    if Current.user
      redirect_to(edit_settings_person_path, notice: "Already logged in") && return
```

Reply:
Oh dang, great catch. And very true about “never” :)